src/library_webgpu.js

```diff
@@ -1543,7 +1543,11 @@ var LibraryWebGPU = {
     var buffer = WebGPU.mgrBuffer.get(bufferId);
     // There is a size limitation for ArrayBufferView. Work around by passing in a subarray
     // instead of the whole heap. crbug.com/1201109
+#if PTHREADS
+    var subarray = HEAPU8.slice(data, data + size);
```

Comment:
It is a Firefox bug that this is needed - we will definitely want to remove this in the future to avoid making an extra copy of the data. Do you know if there is a Firefox bug filed? We'll want to find or file a bug so we can link to it in a comment here, like we have done for the Chrome bugs.

Reply:
oh interesting, that's good to know. I haven't filed one yet, can probably do soon.

Reply:
Thanks! The spec accepts [AllowSharedBufferSource](https://webidl.spec.whatwg.org/#AllowSharedBufferSource) here:
https://gpuweb.github.io/gpuweb/#gpuqueue
```
typedef (ArrayBuffer or SharedArrayBuffer or [AllowShared] ArrayBufferView) AllowSharedBufferSource;
```
and the bug is that Firefox doesn't:
https://searchfox.org/mozilla-central/rev/9c509b8feb28c1e76ad41e65bf9fd87ef672b00f/dom/webidl/WebGPU.webidl#1079,1086

Reply:
Commented on this existing issue here: https://bugzilla.mozilla.org/show_bug.cgi?id=1696216

I'll wait for a first reply before cleaning up this change, unless it takes too long.